File: Server/src/main/java/demo/com/example/testserver/order/dto/OrderDetailRequestDTO.java
package demo.com.example.testserver.order.dto;

import jakarta.validation.constraints.Min;
import jakarta.validation.constraints.NotNull;

public class OrderDetailRequestDTO {

    @NotNull(message = "Product variant ID cannot be null")
    private Integer productVariantId;

    @NotNull(message = "Quantity cannot be null")
    @Min(value = 1, message = "Quantity must be at least 1")
    private Integer quantity;

    // Getters and Setters
    public Integer getProductVariantId() {
        return productVariantId;
    }

    public void setProductVariantId(Integer productVariantId) {
        this.productVariantId = productVariantId;
    }

    public Integer getQuantity() {
        return quantity;
    }

    public void setQuantity(Integer quantity) {
        this.quantity = quantity;
    }
}

File: ecomerge/lib/database/models/order/OrderDetailRequestDTO.dart
class OrderDetailRequestDTO {
  final int productVariantId;
  final int quantity;

  OrderDetailRequestDTO({
    required this.productVariantId,
    required this.quantity,
  });

  Map<String, dynamic> toJson() {
    return {
      'productVariantId': productVariantId,
      'quantity': quantity,
    };
  }
}
